backend/DataLib.js

```diff
@@ -1,147 +1,86 @@
-
-    const startTime = Date.now();
-
-    const retVal = [];
-    for (const key of keys) {
-      const row = termDump.classMap[key];
-      if (row.subject === subject) {
-        retVal.push(key);
-      }
-    }
-
-    // Sort the classes
-    retVal.sort((a, b) => {
-      return parseInt(termDump.classMap[a].classId, 10) - parseInt(termDump.classMap[b].classId, 10);
-    });
-
-    // Turn this into a analytics call when that is working
-    macros.log('send', 'timing', subject, 'subject', Date.now() - startTime);
-
-    return retVal;
-  }
-
-  getClassesInTerm(termId) {
-    if (!this.termDumpMap[termId]) {
+const DataLib = {
+  getClassesInTerm: async (termId) => {
+    const index = `term${termId}`;
+    if (!(await DataLib.hasTerm(termId))) {
       return [];
     }
 
-    return Object.values(this.termDumpMap[termId].classMap);
-  }
+    const searchOutput = await Elastic.search({
+      index: index,
+      size: 10000,
+      body: {
+        query: {
+          match_all: {},
+        },
+      },
+    });
+    return searchOutput.body.hits.hits.map((hit) => {
+      return hit._source.class;
+    });
+  },
 
-  getSectionsInTerm(termId) {
-    if (!this.termDumpMap[termId]) {
+  getSectionsInTerm: async (termId) => {
+    const index = `term${termId}`;
+    if (!(await DataLib.hasTerm(termId))) {
       return [];
     }
+    const searchOutput = await Elastic.search({
+      index: index,
+      size: 10000,
+      body: {
+        query: {
+          match_all: {},
+        },
+      },
+    });
+    return searchOutput.body.hits.hits.reduce((arr, hit) => {
+      return arr.concat(hit._source.sections);
+    }, []);
+  },
 
-    return Object.values(this.termDumpMap[termId].sectionMap);
-  }
-
-  getSubjects(termId) {
+  getSubjects: (termId) => {
     if (!this.termDumpMap[termId]) {
       macros.error("Data lib dosen't have term", termId);
       return null;
     }
 
     return Object.values(this.termDumpMap[termId].subjectMap);
-  }
-
-  hasTerm(termId) {
-    return !!this.termDumpMap[termId];
-  }
-
-  getClassServerDataFromHash(hash) {
-    return this.allClassesMap[hash];
-  }
-
-  getSectionServerDataFromHash(hash) {
-    return this.allSectionsMap[hash];
-  }
-
-  setClass(aClass) {
-    const hash = Keys.getClassHash(aClass);
-
-    if (!this.termDumpMap[aClass.termId]) {
-      macros.error('Cannot set class in non-existent term.');
-      return;
-    }
-
-    this.allClassesMap[hash] = aClass;
-
-    this.termDumpMap[aClass.termId].classMap[hash] = aClass;
-  }
-
-
-  setSection(section) {
-    const hash = Keys.getSectionHash(section);
-
-    if (!this.termDumpMap[section.termId]) {
-      macros.error('Cannot set section in non-existent term.');
-      return;
-    }
-
-    this.allSectionsMap[hash] = section;
-
-    this.termDumpMap[section.termId].sectionMap[hash] = section;
-  }
-}
+  },
+
+  hasTerm: async (termId) => {
+    return Elastic.indices.exists({ index: `term${termId}` });
```

Comment:
And for actually creating this ` `term${termId}` ` identifier - can you use Keys.js - the common way in the codebase to make these identifiers? yaay 🥳🎉

It should end up being something like `neu.edu/201910`